proposals/11551-kubernetes-native-api/TestPlan.md

```diff
@@ -0,0 +1,87 @@
+| (K8s Mode) Try creating an existing pipeline and pipeline verison                                                                    |                                                                                                                                       | Pipeline should not get created with a duplicate name                                                        |
+| (K8s Mode) Run an existing pipeline and get run details and match                                                                    |                                                                                                                                       |                                                                                                              |
+| (K8s Mode) Run all migrated pipeline and match the run details                                                                       |                                                                                                                                       |                                                                                                              |
+| (K8s Mode) Create an experiment and a run                                                                                            |                                                                                                                                       |                                                                                                              |
+
+#### Without migration script
+Create few pipelines in DB mode and migrate it by recreating those via CRs
+
+| **Test Case**                                                                                                                                                                           | **Test Steps**                                          | **Expected Result**                                                                                                                                                 |
+|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| (KFP DB mode) Create pipeline and a pipeline version and re-create the pipeline & version using CRs in K8s mode                                                                         | Match the pipeline and pipeline version (DB vs K8s)     | Pipeline and pipeline version should match exactly except for ID?                                                                                                   |
```

Comment:
Is this using the REST API in both cases and testing the translation of API request to Kubernetes CR?